client/my-sites/post-selector/selector.jsx

```diff
@@ -85,24 +89,31 @@ const PostSelectorPosts = React.createClass( {
 		}, SEARCH_DEBOUNCE_TIME_MS );
 	},
 
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( this.props.query, nextProps.query ) ) {
+			this.setState( {
+				requestedPages: [ 1 ]
```

Comment:
Added `siteId` check in rebased 0e4ec6d. The search refactoring seems a bit more involved so is probably best suited for a separate pull request.